mm2src/coins/sia/src/transaction.rs

```diff
@@ -0,0 +1,1587 @@
+        self.renter_public_key.encode(encoder);
+        self.host_public_key.encode(encoder);
+        encoder.write_u64(self.revision_number);
+        self.renter_signature.encode(encoder);
+        self.host_signature.encode(encoder);
+    }
+}
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct V2FileContractElement {
+    #[serde(flatten)]
+    pub state_element: StateElement,
+    pub v2_file_contract: V2FileContract,
+}
+
+impl Encodable for V2FileContractElement {
+    fn encode(&self, encoder: &mut Encoder) {
+        self.state_element.encode(encoder);
+        self.v2_file_contract.encode(encoder);
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct FileContractRevisionV2 {
+    pub parent: V2FileContractElement,
+    pub revision: V2FileContract,
+}
+
+impl FileContractRevisionV2 {
+    pub fn with_nil_sigs(&self) -> FileContractRevisionV2 {
+        FileContractRevisionV2 {
+            revision: self.revision.with_nil_sigs(),
+            ..self.clone()
+        }
+    }
+}
+
+impl Encodable for FileContractRevisionV2 {
+    fn encode(&self, encoder: &mut Encoder) {
+        self.parent.encode(encoder);
+        self.revision.encode(encoder);
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct Attestation {
+    pub public_key: PublicKey,
+    pub key: String,
+    pub value: Vec<u8>,
+    pub signature: Signature,
+}
+
+impl Encodable for Attestation {
+    fn encode(&self, encoder: &mut Encoder) {
+        self.public_key.encode(encoder);
+        encoder.write_string(&self.key);
+        encoder.write_len_prefixed_bytes(&self.value);
+        self.signature.encode(encoder);
+    }
+}
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct StorageProof {
+    pub parent_id: FileContractID,
+    pub leaf: HexArray64,
+    pub proof: Vec<H256>,
+}
+
+type SiafundOutputID = H256;
+type FileContractID = H256;
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct FileContractRevision {
+    pub parent_id: FileContractID,
+    pub unlock_condition: UnlockCondition,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SiafundInputV1 {
+    pub parent_id: SiafundOutputID,
+    pub unlock_condition: UnlockCondition,
+    pub claim_address: Address,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub enum ResolutionType {
+    Renewal,
+    StorageProof,
+    Expiration,
+    Finalization,
+}
+
+#[derive(Clone, Debug, Serialize, PartialEq)]
+pub struct V2FileContractResolution {
+    pub parent: V2FileContractElement,
+    #[serde(rename = "type")]
+    pub resolution_type: ResolutionType,
+    pub resolution: V2FileContractResolutionWrapper,
+}
+
+impl Encodable for V2FileContractResolution {
+    fn encode(&self, _encoder: &mut Encoder) { todo!() }
```

Comment:
Not sure if you forgot about this todo or you are leaving it for a future implementation. Same for other `todo!()` usages. Just a reminder in case you forgot about it :)

Reply:
We have no technical need to implement this. I will circle back in later implementations to either make sure we never attempt to encode this object or fully port the Go logic.